artifact/artifact.sh

```diff
@@ -1,48 +1,89 @@
+            if [[ ",${RX_ENABLED_BRANCHES:-}," = *",${BRANCH},"* ]]; then RX_ENABLED=true; fi
+            DOCKER_BUILDKIT=1 docker build --tag "${DOCKER_REPO}" --build-arg ROLLBAR_POST_SERVER_TOKEN="${ROLLBAR_POST_SERVER_TOKEN:-}" --build-arg LAUNCHDARKLY_CLIENT_TOKEN="${LAUNCHDARKLY_CLIENT_TOKEN:-}" --build-arg REACT_APP_GAID="${REACT_APP_GAID:-}" --build-arg RX_ENABLED="${RX_ENABLED:-}" --build-arg TRAVIS_COMMIT="${COMMIT:-}" .
         else
+            # Build other images
             docker build --tag "${DOCKER_REPO}" .
         fi
 
-        if [ "${TRAVIS_BRANCH:-}" == "master" ] && [ "${TRAVIS_PULL_REQUEST_BRANCH:-}" == "" ]; then
+        if [ "${BRANCH:-}" == "master" ] && [ $IS_PULL_REQUEST == false ]; then
+            # Push master branch image
             docker push "${DOCKER_REPO}"
         fi
-        if [ -n "${TRAVIS_TAG:-}" ]; then
-            docker tag "${DOCKER_REPO}" "${DOCKER_REPO}:${TRAVIS_TAG}"
-            docker push "${DOCKER_REPO}:${TRAVIS_TAG}"
+
+        if [ -n "${TAG:-}" ]; then
+            # Push git tag image
+            docker tag "${DOCKER_REPO}" "${DOCKER_REPO}:${TAG}"
+            docker push "${DOCKER_REPO}:${TAG}"
         fi
-        if [ -n "${TRAVIS_BRANCH:-}" ] && [ -n "${TRAVIS_COMMIT:-}"  ]; then
-            if [ -n "${TRAVIS_PULL_REQUEST_BRANCH}" ]
+
+        if [ -n "$BRANCH" ] && [ -n "$COMMIT" ]; then
+            if [ $IS_PULL_REQUEST == true ]
             then
-                BRANCH=$(echo -n ${TRAVIS_PULL_REQUEST_BRANCH} | tr / -)
+                TAG="PR-${PR_NUMBER}-${COMMIT}"
             else
-                BRANCH=$(echo -n ${TRAVIS_BRANCH} | tr / -)
+                TAG=$(echo -n ${BRANCH} | tr / -)-${COMMIT}
             fi
-            docker tag "${DOCKER_REPO}" "${DOCKER_REPO}:${BRANCH}-${TRAVIS_COMMIT}"
-            docker push "${DOCKER_REPO}:${BRANCH}-${TRAVIS_COMMIT}"
+            # Push commit and timestamp images
+            docker tag "${DOCKER_REPO}" "${DOCKER_REPO}:${TAG}"
+            docker push "${DOCKER_REPO}:${TAG}"
 
             TIMESTAMP=$(date +%s)
-            docker tag "${DOCKER_REPO}" "${DOCKER_REPO}:${BRANCH}-${TRAVIS_COMMIT}-${TIMESTAMP}"
-            docker push "${DOCKER_REPO}:${BRANCH}-${TRAVIS_COMMIT}-${TIMESTAMP}"
+            docker tag "${DOCKER_REPO}" "${DOCKER_REPO}:${TAG}-${TIMESTAMP}"
+            docker push "${DOCKER_REPO}:${TAG}-${TIMESTAMP}"
 
-            docker tag "${DOCKER_REPO}" "${DOCKER_REPO}:${BRANCH}-latest"
-            docker push "${DOCKER_REPO}:${BRANCH}-latest"
+            # Push branch latest image
+            docker tag "${DOCKER_REPO}" "${DOCKER_REPO}:$(echo -n ${BRANCH} | tr / -)-latest"
```

Comment:
We don't want to push the `latest` using the `branch` for PR builds. This will just result in the same two different builds with an indeterminant source (is it from the feature branch build or the PR build). For a PR build, we'd want to use the same `PR-123` form with the `-lastest` suffix.

Reply:
So, you may want to remove the `-${COMMIT}` from the definition of `TAG` above and apply that manually. Then you can use `${TAG}` here instead of the branch.